### PR TITLE
feat(taiko-client): compatible changes for `lastProposedIn`

### PR DESCRIPTION
**Pull request by YoGhurt111**

### CODE REVIEW

File: packages/taiko-client/pkg/rpc/utils.go [hunk 1]
```diff
@@ -31,6 +32,7 @@ var (
 		syscall.SIGTERM,
 		syscall.SIGQUIT,
 	}
+	ErrSlotBMarshal = errors.New("abi: cannot marshal in to go type: length insufficient 160 require 192")
```
Comment (mask-pp) [inline packages/taiko-client/pkg/rpc/utils.go:35]:
```suggestion
	ErrSlotBMarshal = errors.New("abi: cannot marshal into go type: length insufficient 160 require 192")
```

**Comment by dantaik** [comment]
Will this change be compatible with 1.10.0 (without this new field)?

**Comment by YoGhurt111** [comment]
> Will this change be compatible with 1.10.0 (without this new field)?

Yes, it is compatible with or without the new field.

File: packages/taiko-client/pkg/rpc/utils.go (hunk 1 above)
Comment (YoGhurt111) [reply packages/taiko-client/pkg/rpc/utils.go:35]: It is copied from the error msg, so no need to change.

**Review by mask-pp — APPROVED** [review]

**Comment by mask-pp** [comment]
Need to fix the failed ci.

**Review by dantaik — APPROVED** [review]

File: packages/taiko-client/pkg/rpc/utils.go [hunk 2]
```diff
@@ -63,9 +65,22 @@ func GetProtocolStateVariables(
 	opts.Context, cancel = CtxWithTimeoutOrDefault(opts.Context, defaultTimeout)
 	defer cancel()
 
+	var slotBV1 bindings.TaikoDataSlotBV1
 	slotA, slotB, err := taikoL1Client.GetStateVariables(opts)
 	if err != nil {
-		return nil, err
+		if errors.Is(err, ErrSlotBMarshal) {
+			slotA, slotBV1, err = taikoL1Client.GetStateVariablesV1(opts)
```
Comment (davidtaikocha) [inline packages/taiko-client/pkg/rpc/utils.go:72]: Shall we also check the error here?

File: packages/taiko-client/pkg/rpc/utils.go [hunk 3]
```diff
@@ -63,9 +65,22 @@ func GetProtocolStateVariables(
...
+				NumBlocks:           slotBV1.NumBlocks,
+				LastVerifiedBlockId: slotBV1.LastVerifiedBlockId,
+				ProvingPaused:       slotBV1.ProvingPaused,
+				LastProposedIn:      nil,
+				LastUnpausedAt:      slotBV1.LastUnpausedAt,
+			}
+		}
+		if err != nil {
+			return nil, err
+		}
```
Comment (davidtaikocha) [inline packages/taiko-client/pkg/rpc/utils.go:83]: Seems there are still some places are using the raw `c.TaikoL1.GetStateVariables`, lets also change them to use the method here?

Comment (YoGhurt111) [reply packages/taiko-client/pkg/rpc/utils.go:83]: Yes, but we also see the raw `GetStateVariables ` used in Indexer.
Maybe we should make Jeff noticed?

Comment (davidtaikocha) [reply packages/taiko-client/pkg/rpc/utils.go:83]: > Yes, but we also see the raw `GetStateVariables ` used in Indexer. Maybe we should make Jeff noticed?

Correct, cc @cyberhorsey

File: packages/taiko-client/pkg/rpc/utils.go (hunk 2 above)
Comment (YoGhurt111) [reply packages/taiko-client/pkg/rpc/utils.go:72]: I think there is no need.
If we fail both ways, just throw an error!

Comment (davidtaikocha) [reply packages/taiko-client/pkg/rpc/utils.go:72]: If we don't check the `err` here, `slotBV1` / `slotA` will be `nil` and the code below will panic?

**Comment by YoGhurt111** [comment]
I tested this pr with reverting https://github.com/taikoxyz/taiko-mono/pull/18379.
All ci passed.

File: packages/taiko-client/pkg/rpc/utils.go (hunk 2 above)
Comment (YoGhurt111) [reply packages/taiko-client/pkg/rpc/utils.go:72]: Yes, fixed

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**